### PR TITLE
Website 78 replace moment dependency

### PR DESCRIPTION
**Pull request by SalazarJosh**
# Overview
Moment.js is no longer supported and was built on an old era of the JS ecosystem. Replacement options included date-fns _[[website](https://date-fns.org/), [github](https://github.com/date-fns/date-fns)]_ and day.js _[[website](https://day.js.org/), [github](https://github.com/iamkun/dayjs)]_. I ultimately decided to go with date-fns since it appears to have a larger user base and more support.

This PR should move everything from moment to date-fns. I'm not 100% sure what the locale formatting of parseISO is doing despite my best efforts to understand it. My goal was to create the same outputs from moment with date-fns and I believe I've done that.

All changed components:
- src/components/panels/events-and-exhibits-panel.js
- src/components/panels/featured-and-latest-news.js
- src/components/panels/hours-lite-panel.js
- src/components/panels/hours-panel.js
- src/components/todays-hours.js
- src/templates/news-landing.js
- src/templates/event.js
- src/templates/news.js
- src/utils/events.js
- src/utils/hours.js

[date-fns documentation](https://date-fns.org/docs/Getting-Started)

> This pull request resolves/closes/fixes [WEBSITE-78](https://mlit.atlassian.net/browse/WEBSITE-78).

Let's remove moment with the June website updates.

## Anything else?
[All conversion tracking was done in this document.](https://docs.google.com/spreadsheets/d/1F9PTjNtmXse1CDv4SM8_KAIWGRv4ezkeD-7yNa8D8MQ/edit#gid=289530058) I linked the second tab (Pre Deploy Testing Status (Phase 2)) where I share some additional notes about the changes and the live site with an example of where that component appears. The first tab is my development tracking document for myself.

## Testing
List instructions on how to test the pull request. Some examples:

- Make sure the PR is consistent in these browsers:
  - [x] Chrome
  - [x] Firefox
  - [ ] Safari
  - [x] Edge (the assignee was not able to test the pull request in this browser)
- Run accessibility tests:
  - [ ] WAVE
  - [ ] ARC Toolkit
  - [ ] axe DevTools

### CODE REVIEW

**Review by SalazarJosh — COMMENTED** [review]
@erinesullivan It looks like the package-lock file is causing issues. Apologies for my ignorance here, should the package-lock file even be committed? I'm not sure what I'm doing wrong here. That's the bulk of the 1,200 changed lines and it's really muddying up this PR.

File: src/utils/hours.js [hunk 1]
```diff
@@ -90,12 +90,14 @@ export function findHoursSetByNodeForNow({ node, now }) {
...
-    const start = set.field_date_range.value;
-    const end = set.field_date_range.end_value;
+    // Start and end of hour set range.
+    const start = parseISO(set.field_date_range.value);
+    const end = parseISO(set.field_date_range.end_value);
 
     // Check if "now" is within the range including start and end days.
-    return now.isSameOrAfter(start, 'day') && now.isSameOrBefore(end, 'day');
+    return (
+      isSameDay(now, start) || isSameDay(now, end) || (now > start && now < end)
```
Comment (erinesullivan) [inline src/utils/hours.js:99]:
```suggestion
      !(now < start || now > end)
```
I believe this line could be simplified to check if now is neither before the start or after the end.

File: src/utils/hours.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import * as moment from 'moment';
+import { isSameDay, parseISO, getDay, parse, format } from 'date-fns';
```
Comment (erinesullivan) [inline src/utils/hours.js:1]:
```suggestion
import { parseISO, getDay, parse, format } from 'date-fns';
```
`isSameDay` is not needed if using the simplified code below.

File: src/utils/hours.js [hunk 3]
```diff
@@ -90,12 +90,14 @@ export function findHoursSetByNodeForNow({ node, now }) {
       return null;
     }
 
-    // Star and end of hour set range.
-    const start = set.field_date_range.value;
-    const end = set.field_date_range.end_value;
+    // Start and end of hour set range.
```
Comment (erinesullivan) [inline src/utils/hours.js:93]: 😄 Good job quality controlling the comments!

**Review by erinesullivan — CHANGES_REQUESTED** [review]
👏🏻 Great pull request description. Way to set the bar for your first PR!

I tested the branch in Chrome, Firefox, Safari, and Edge. I did not see anything different from the live site, which is exactly what we want. Awesome job!

Two changes:
- I found a line in `src/utils/hours.js` that could be simplified.
- Since ESLint is set up for the repository, it should become a common practice to go through all the files in every open PR and clean them up. There are a lot of errors that can be fixed automatically in the 10 edited files (excluding `package` related files), but a lot remain even after the fixes. Let me know if you need help with them.
```
npx eslint src/components/panels/events-and-exhibits-panel.js src/components/panels/featured-and-latest-news.js src/components/panels/hours-lite-panel.js src/components/panels/hours-panel.js src/components/todays-hours.js src/templates/event.js src/templates/news-landing.js src/templates/news.js src/utils/events.js src/utils/hours.js --fix
```

**Pull request closed without merging**